src/game/ingame_menu.c

```diff
@@ -558,25 +730,64 @@ void print_credits_string(s16 x, s16 y, const u8 *str) {
                 G_TX_CLAMP, 3, G_TX_NOLOD, G_TX_CLAMP, 3, G_TX_NOLOD);
     gDPSetTileSize(gDisplayListHead++, G_TX_RENDERTILE, 0, 0, (8 - 1) << G_TEXTURE_IMAGE_FRAC, (8 - 1) << G_TEXTURE_IMAGE_FRAC);
 
-    while (str[strPos] != GLOBAR_CHAR_TERMINATOR) {
-        switch (str[strPos]) {
-            case GLOBAL_CHAR_SPACE:
-                curX += 4;
-                break;
-            default:
-                gDPPipeSync(gDisplayListHead++);
-                gDPSetTextureImage(gDisplayListHead++, G_IM_FMT_RGBA, G_IM_SIZ_16b, 1, fontLUT[str[strPos]]);
-                gDPLoadSync(gDisplayListHead++);
-                gDPLoadBlock(gDisplayListHead++, G_TX_LOADTILE, 0, 0, 8 * 8 - 1, CALC_DXT(8, G_IM_SIZ_16b_BYTES));
-                gSPTextureRectangle(gDisplayListHead++, curX << 2, curY << 2, (curX + 8) << 2,
-                                    (curY + 8) << 2, G_TX_RENDERTILE, 0, 0, 1 << 10, 1 << 10);
-                curX += 7;
-                break;
+    while (str[strPos] != '\0') {
+        if (fontLUT[str[strPos] - ' '].texture != NULL) {
+            gDPPipeSync(gDisplayListHead++);
+            gDPSetTextureImage(gDisplayListHead++, G_IM_FMT_RGBA, G_IM_SIZ_16b, 1, fontLUT[str[strPos] - ' '].texture);
+            gDPLoadSync(gDisplayListHead++);
+            gDPLoadBlock(gDisplayListHead++, G_TX_LOADTILE, 0, 0, 8 * 8 - 1, CALC_DXT(8, G_IM_SIZ_16b_BYTES));
+            gSPTextureRectangle(gDisplayListHead++, curX << 2, curY << 2, (curX + 8) << 2,
+                                (curY + 8) << 2, G_TX_RENDERTILE, 0, 0, 1 << 10, 1 << 10);
         }
+
+        curX += fontLUT[str[strPos] - ' '].kerning;
         strPos++;
     }
 }
 
+/**
+ * Variants of the above that allow for text alignment.
+ */
+void print_generic_string_aligned(s16 x, s16 y, char *str, u32 alignment) {
+    s32 strLength = get_string_length(str, main_font_lut, &main_font_utf8_lut);
```

Comment:
true i should have calculated this before left align. newlines aren't really supported by this, im not sure how i would do it. i can't do "proper" center / right alignment because i cant affect the starting position of each line, only the first one

Reply:
i meant after checking left align*